### DISCUSSION TITLE
PPTX Import with Standard Layout Support

### DISCUSSION THREAD

**Discussion by frettig-radiox** [jgm/pandoc#9643] [Ideas]
I suggest creating a PPTX importer for Pandoc designed to maintain standard presentation layouts throughout the conversion process ([like the export does](https://pandoc.org/MANUAL.html#powerpoint-layout-choice)). This implies that the feature would support the consistent export and re-import of standard layouts. Feedback and collaboration from the community would be invaluable to address this functionality gap and refine implementation strategies.

Regards,
Frank

**Comment by jgm** [comment]
That sounds good, but the main issue is that the pptx format is really gnarly, and the developer who created the pptx writer and is familiar with its details is no longer active.  If someone who is familiar with Haskell wants to get intimately familiar with pptx format, then there are many places where improvements could be made.

**Comment by frettig-radiox** [comment]
Hi,
the format itself is not the problem. I am reasonably familiar with it. How much time do you estimate would be needed to get accustomed to Haskell and the existing source code? I think the docx importer would be a good starting point.

**Reply by jgm** [reply]
It's impossible to estimate that, it depends on the person.  You could look at the docx reader's source code for a start, but that won't teach you Haskell.  My guess is that there would be some possibilities for factoring out code common to docx and pptx readers.

**Comment by ElDuderino321** [comment]
I've just come to appreciate the ability to use the pptx writer to generate my presentations (which I need to have as pptx). It would be awesome if the functionality could be expanded, both with imports and templating.

(If I try to edit the default pptx template by exporting and editing the styles and fields in LibreOffice, the saved version is unusable for pandoc, and layouts are not retained. But that is a LibreOffice issue. Right now, I create a bare bone version which I further tweak and style in google slides.)
